restcomm/restcomm.testsuite/src/test/java/org/restcomm/connect/testsuite/http/CallsEndpointTest.java

```diff
@@ -59,57 +61,100 @@ public void getCallsList() {
-                adminAccountSid, adminAuthToken, 2, null, true);
+                adminAccountSid, adminAuthToken, 2, null, null, true);
         JsonArray secondPageCallsArray = secondPage.get("calls").getAsJsonArray();
-        assertTrue(secondPageCallsArray.size() == 50);
-        assertTrue(secondPage.get("start").getAsInt() == 100);
-        assertTrue(secondPage.get("end").getAsInt() == 149);
+        assertEquals(secondPageCallsArray.size(), 50);
+        assertEquals(secondPage.get("start").getAsInt(), 100);
+        assertEquals(secondPage.get("end").getAsInt(), 149);
 
         JsonObject lastPage = (JsonObject) RestcommCallsTool.getInstance().getCalls(deploymentUrl.toString(), adminAccountSid,
-                adminAuthToken, firstPage.get("num_pages").getAsInt(), null, true);
+                adminAuthToken, firstPage.get("num_pages").getAsInt(), null, null, true);
         JsonArray lastPageCallsArray = lastPage.get("calls").getAsJsonArray();
-        assertTrue(lastPageCallsArray.get(lastPageCallsArray.size() - 1).getAsJsonObject().get("sid").getAsString()
-                .equals("CAe803a594ac1649d98855eafc7535ed41"));
-        assertTrue(lastPageCallsArray.size() == 48);
-        assertTrue(lastPage.get("start").getAsInt() == 400);
-        assertTrue(lastPage.get("end").getAsInt() == 448);
+        assertEquals(lastPageCallsArray.get(lastPageCallsArray.size() - 1).getAsJsonObject().get("sid").getAsString(),
+                "CA22ac5fd16a954c339f33519bdfe4af78");
+        assertEquals(lastPageCallsArray.size(), 48);
+        assertEquals(lastPage.get("start").getAsInt(), 400);
+        assertEquals(lastPage.get("end").getAsInt(), 448);
 
-        assertTrue(totalSize == 448);
+        assertEquals(totalSize, 448);
+    }
+
+    @Test
+    public void getCallsListUsingSorting() {
+        // Provide both sort field and direction
+        // Provide ascending sorting and verify that the first row is indeed the earliest one
+/*
+        JsonObject response1 = RestcommCallsTool.getInstance().getCalls(deploymentUrl.toString(),
+                adminAccountSid, adminAuthToken, 0, 10, "date_created:asc", true);
+        assertEquals(((JsonObject)response1.get("calls").getAsJsonArray().get(0)).get("date_created").getAsString(), "Fri, 5 Jul 2013 22:15:53 +0300");
+
+        // Provide only sort field; direction should default to desc
+        JsonObject response2 = RestcommCallsTool.getInstance().getCalls(deploymentUrl.toString(),
+                adminAccountSid, adminAuthToken, 0, 10, "date_created", true);
+        assertEquals(((JsonObject)response2.get("calls").getAsJsonArray().get(0)).get("date_created").getAsString(), "Tue, 31 May 2016 16:20:22 +0300");
+*/
+
+        JsonObject response3 = RestcommCallsTool.getInstance().getCalls(deploymentUrl.toString(),
+                adminAccountSid, adminAuthToken, 0, 10, "date_created:desc", true);
```

Comment:
is this the expected name "data_created"?
also ,lets try to check DAO behavior on unit tests first..

Reply:
Sure, will check unit tests too